src/NServiceBus.AzureStoragePersistence/Subscriptions/Azure/AzureSubscriptionStorage.cs

```diff
@@ -1,100 +1,103 @@
 ﻿namespace NServiceBus.Unicast.Subscriptions
 {
+    using System;
     using System.Collections.Generic;
     using System.Linq;
-    using Azure;
-    using MessageDrivenSubscriptions;
+    using System.Text;
     using Microsoft.WindowsAzure.Storage;
+    using Microsoft.WindowsAzure.Storage.RetryPolicies;
     using Microsoft.WindowsAzure.Storage.Table;
-    using Microsoft.WindowsAzure.Storage.Table.DataServices;
+    using Microsoft.WindowsAzure.Storage.Table.Queryable;
+    using NServiceBus.Azure;
+    using NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions;
 
     /// <summary>
-    /// 
     /// </summary>
     public class AzureSubscriptionStorage : ISubscriptionStorage
     {
-        CloudTableClient client;
+        private readonly string subscriptionTableName;
+        private CloudTableClient client;
 
         /// <summary>
-        /// 
         /// </summary>
-        /// <param name="account"></param>
-        public AzureSubscriptionStorage(CloudStorageAccount account)
+        /// <param name="subscriptionTableName">Table name used to store subscription information</param>
+        /// <param name="subscriptionConnectionString">Subscription connection string</param>
+        public AzureSubscriptionStorage(string subscriptionTableName, string subscriptionConnectionString)
         {
-            client = account.CreateCloudTableClient();           
+            this.subscriptionTableName = subscriptionTableName;
+            var account = CloudStorageAccount.Parse(subscriptionConnectionString);
+
+            client = account.CreateCloudTableClient();
+            client.DefaultRequestOptions = new TableRequestOptions
+            {
+                RetryPolicy = new ExponentialRetry()
+            };
         }
 
         void ISubscriptionStorage.Subscribe(Address address, IEnumerable<MessageType> messageTypes)
         {
-            using (var context = new SubscriptionServiceContext(client))
```

Comment:
ok this is my ignorance showing. if GetTableReference bundles what is the trigger for the budle to be sent? since there is nothing i can see that indicates scope